### PR TITLE
[stable/nginx-ingress] Add prometheus rule

### PR DESCRIPTION
**Pull request by mogaal**
#### What this PR does / why we need it:

Sometimes we want to trigger alerts depending of certain metrics. For instance: Too many 500s, latency thresholds, etc

This PR adds the possibility of setting prometheusRules (for Prometheus Operator). It goes hand in hand with serviceMonitor (which this chart already implements) but it's mainly designed for alerting (rules in prometheus that most probably are going be picked up by alertmanager). 

I tried to keep the same approach the chart has for serviceMonitor, which is quite standard. 

#### Special notes for your reviewer:

You'll have prometheus-operator running and set 

```yaml
    serviceMonitor:
      enabled: true
```

Ensure you're getting metrics inside prometheus and later set the prometheusRules, have a look at the examples I included inside the default values in case you want to test some of the alerts.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/nginx-ingress/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: nginx-ingress
-version: 1.15.1
+version: 1.15.2
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/Chart.yaml]: please bump the minor version to `1.16.0` as this is a new feature

File: stable/nginx-ingress/README.md [hunk 2]
```diff
@@ -138,6 +138,10 @@ Parameter | Description | Default
 `controller.metrics.serviceMonitor.additionalLabels` | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus | `{}`
 `controller.metrics.serviceMonitor.namespace` | namespace where servicemonitor resource should be created | `the same namespace as nginx ingress`
 `controller.metrics.serviceMonitor.honorLabels` | honorLabels chooses the metric's labels on collisions with target labels. | `false`
+`controller.metrics.prometheusRule.enabled` | Set this to `true` to create prometheusRules for Prometheus operator | `false`
+`controller.metrics.prometheusRule.additionalLabels` | Additional labels that can be used so prometheusRules will be discovered by Prometheus | `{}`
+`controller.metrics.prometheusRule.namespace` | namespace where prometheusRules resource should be created | `the same namespace as nginx ingress`
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/README.md:143]: namespace is handled by helm and doesn't need to be included in our templates.

File: stable/nginx-ingress/templates/controller-prometheusrules.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.controller.metrics.enabled .Values.controller.metrics.prometheusRule.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ template "nginx-ingress.controller.fullname" . }}
+  {{- if .Values.controller.metrics.prometheusRule.namespace }}
+  namespace: {{ .Values.controller.metrics.prometheusRule.namespace }}
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/templates/controller-prometheusrules.yaml:7]: namespace is handled by helm and doesn't need to be included in our templates.

**Review by ChiefAlexander — CHANGES_REQUESTED** [review]
A few changes. Otherwise this looks great!
/ok-to-test

File: stable/nginx-ingress/Chart.yaml (hunk 1 above)
Comment (mogaal) [reply stable/nginx-ingress/Chart.yaml]: Ohh cool, just did it. Thanks

File: stable/nginx-ingress/README.md (hunk 2 above)
Comment (mogaal) [reply stable/nginx-ingress/README.md:143]: Yes, I understand namespace is handled by helm but for most of the situation you will require to specify another one different and this gives you the possibility. Actually, it's my case: we have the monitoring/alerting system in another namespace different than the one where nginx-ingress is deployed. 

You're already implementing the same approach in this chart, check `templates/controller-servicemonitor.yaml` ( https://github.com/helm/charts/blob/master/stable/nginx-ingress/templates/controller-servicemonitor.yaml#L6-L8 )

File: stable/nginx-ingress/templates/controller-prometheusrules.yaml (hunk 3 above)
Comment (mogaal) [reply stable/nginx-ingress/templates/controller-prometheusrules.yaml:7]: Same as above, check templates/controller-servicemonitor.yaml ( https://github.com/helm/charts/blob/master/stable/nginx-ingress/templates/controller-servicemonitor.yaml#L6-L8 )

**Comment by ChiefAlexander** [comment]
Thank you for the PR!
/lgtm

**Comment by lephleg** [comment]
@mogaal Is by any chance this PR related to an upgrade error I'm receiving recently: https://github.com/helm/charts/issues/16641

Please advice, as this is already deployed to production.

**Pull request merged**